### PR TITLE
Bugfix/update map origin with home position

### PR DESCRIPTION
**Pull request by natmol**
Updates the `map_origin` in the callback `home_position_cb` after new home has been updated from the FMU over the mavtlink message [HOME_POSITION](https://mavlink.io/en/messages/common.html#HOME_POSITION) and in the ros topic `/mavros/home_position/home`

### CODE REVIEW

File: mavros/src/plugins/global_position.cpp [hunk 1]
```diff
@@ -89,7 +90,7 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 
 		// home position subscriber to set "map" origin
 		// TODO use UAS
-		hp_sub = gp_nh.subscribe("home", 10, &GlobalPositionPlugin::home_position_cb, this);
+		hp_sub = nh.subscribe("/mavros/home_position/home", 10, &GlobalPositionPlugin::home_position_cb, this);
```
Comment (vooon) [inline mavros/src/plugins/global_position.cpp:93]: No! Node mustn't use absolute paths.

File: mavros/src/plugins/global_position.cpp [hunk 2]
```diff
@@ -98,16 +99,17 @@ class GlobalPositionPlugin : public plugin::PluginBase {
...
-				make_handler(&GlobalPositionPlugin::handle_gps_raw_int),
-				// GPS_STATUS: there no corresponding ROS message, and it is not supported by APM
-				make_handler(&GlobalPositionPlugin::handle_global_position_int),
-				make_handler(&GlobalPositionPlugin::handle_gps_global_origin),
-				make_handler(&GlobalPositionPlugin::handle_lpned_system_global_offset)
+			make_handler(&GlobalPositionPlugin::handle_gps_raw_int),
+			// GPS_STATUS: there no corresponding ROS message, and it is not supported by APM
+			make_handler(&GlobalPositionPlugin::handle_global_position_int),
+			make_handler(&GlobalPositionPlugin::handle_gps_global_origin),
+			make_handler(&GlobalPositionPlugin::handle_lpned_system_global_offset)
```
Comment (vooon) [inline mavros/src/plugins/global_position.cpp:106]: Do not change style.

**Review by vooon — CHANGES_REQUESTED** [review]

File: mavros/src/plugins/global_position.cpp (hunk 1 above)
Comment (natmol) [reply mavros/src/plugins/global_position.cpp:93]: The Mavlink HOMEPOSITION message is handled by the home_position.cpp plugin. 

image

This gets published in the topic `/mavros/home_position/home`. 

Is it desired to create a handler in the global_position.cpp plugin for the same Mavlink topic?

File: mavros/src/plugins/global_position.cpp (hunk 2 above)
Comment (natmol) [reply mavros/src/plugins/global_position.cpp]: Done. Style changed with uncrustify

File: mavros/src/plugins/global_position.cpp (hunk 1 above)
Comment (vooon) [reply mavros/src/plugins/global_position.cpp:93]:
```cpp
NodeHandle priv_nh("~");

sub = priv_nh.subscribe("home_position/home", ...);
```

Comment (vooon) [reply mavros/src/plugins/global_position.cpp:93]: Or even:
```
NodeHandle hp_nh("~home_position");

sub = hp_nh.subscribe("home", ...);
```

**Review by vooon — APPROVED** [review]
LGTM, thanks!

**Pull request merged**